internal/controller/gcpinstall/run.go

```diff
@@ -52,23 +70,85 @@ https://cloud.google.com/iam/docs/creating-managing-service-accounts`,
+				command.PassState: "This Flightcrew IAM role already exists.",
+				command.FailState: "No IAM role found. Next step is to create one.",
+			},
+		})
+		cmd.Replace(replacer)
+		return cmd
+	}
+	newCreateIAMRole := func(replacer *strings.Replacer, skipIfSucceed *command.Model) *command.Model {
+		cmd := command.NewWriteModel(command.Opts{
+			SkipIfSucceed: skipIfSucceed,
+			Description:   "This command creates a ${PERMISSIONS} IAM role from `${FILE}` for the Flightcrew VM to access configs and monitoring data.\n\nhttps://cloud.google.com/iam/docs/understanding-custom-roles",
+			Command: `gcloud iam roles create ${ROLE} \${PROJECT_OR_ORG_FLAG}
+	--file=${FILE}`,
+		})
+		cmd.Replace(replacer)
+		return cmd
+	}
+	newAttachPolicy := func(replacer *strings.Replacer, skipIfSucceed *command.Model) *command.Model {
+		cmd := command.NewWriteModel(command.Opts{
+			SkipIfSucceed: skipIfSucceed,
+			Description: `This command attaches the ${PERMISSIONS} IAM role to Flightcrew's service account, which will give the attached permissions to a new VM.
 
 https://cloud.google.com/iam/docs/granting-changing-revoking-access`,
-			Command: `gcloud projects add-iam-policy-binding "${GOOGLE_PROJECT_ID}" \
+			Command: `gcloud projects get-iam-policy-binding "${GOOGLE_PROJECT_ID}" \
 	--member=serviceAccount:"${SERVICE_ACCOUNT}@${GOOGLE_PROJECT_ID}.iam.gserviceaccount.com" \
-	--role="${PROJECT_OR_ORG_SLASH}/roles/${IAM_ROLE}" \
+	--role="${PROJECT_OR_ORG_SLASH}/roles/${ROLE}" \
 	--condition=None`,
-		}),
+		})
+		cmd.Replace(replacer)
+		return cmd
+	}
+
+	commands := make([]*command.Model, 0)
+
+	readReplacer := strings.NewReplacer(
+		"${ROLE}", args[keyIAMRoleRead],
+		"${FILE}", args[keyIAMFileRead],
+		"${PERMISSIONS}", constants.Read)
+	checkReadIAMRole := newCheckIAMRole(readReplacer)
+	commands = append(commands, checkReadIAMRole)
+	commands = append(commands, newCreateIAMRole(readReplacer, checkReadIAMRole))
+	commands = append(commands, newAttachPolicy(readReplacer, checkReadIAMRole))
+
+	if args[keyPermissions] == constants.Write {
+		writeReplacer := strings.NewReplacer(
+			"${ROLE}", args[keyIAMRoleWrite],
+			"${FILE}", args[keyIAMFileWrite],
+			"${PERMISSIONS}", constants.Write)
+		checkWriteIAMRole := newCheckIAMRole(writeReplacer)
+		commands = append(commands, checkWriteIAMRole)
+		commands = append(commands, newCreateIAMRole(writeReplacer, checkWriteIAMRole))
+		commands = append(commands, newAttachPolicy(writeReplacer, checkWriteIAMRole))
+	}
+
+	return commands
+}
+
+func getVMCommands(args map[string]string) []*command.Model {
+	checkVMExists := command.NewReadModel(command.Opts{
+		Description: "Check if a Flightcrew VM already exists or needs to be created.",
+		Command: `gcloud compute instances list --format="csv(NAME,EXTERNAL_IP,STATUS)" \${PROJECT_OR_ORG_FLAG}
+	--zones=${ZONE} | awk -F "," "/${VIRTUAL_MACHINE}/ {print f(2), f(3)} function f(n){return (\$n==\"\" ? \"null\" : \$n)}" | [ $(wc -c) -gt "0" ]`,
```

Comment:
I can make this much easier with the GCP instancesClient if we want to do that at some point. But I think it's been easier to use gcloud for now

Reply:
I can do that as well, just need to refactor the `command.Model` so that it can be either a function or a bash command. Probably better as a followup PR to make it easier to review.

Reply:
Oh yeah, we don't need to do this now. Was just noting if this was annoying that it's possible to fix